pkg/scheduler/framework/types.go

```diff
@@ -581,7 +585,7 @@ func (n *NodeInfo) RemovePod(pod *v1.Pod) error {
 			return nil
 		}
 	}
-	return fmt.Errorf("no corresponding pod %s in pods of node %s", pod.Name, n.node.Name)
+	return fmt.Errorf("no corresponding pod %s in pods of node %s", pod.Name, n.name)
```

Comment:
getting here means there is another bug, so before we decide whether or not it is worth carrying the node name in nodeinfo, lets debug why we are here in the first place. 

/cc @alculquicondor

Reply:
This would only happen if the Pod was already removed, which in turn would be when 2 Pod delete events are received.
But we can use pod.Spec.NodeName.
Unless this is a bug that we already solved (can't find the PR).

Reply:
I found the precedents: #89553 (not merged) and #89908 (merged to 1.19 and 1.18), but also reverted in #93938

So, it's important to know which version the OP was running to see if this is still an issue

Reply:
is there a legitimate case where we get two delete events for a pod?